### ISSUE TITLE
``no-member`` check does not honor ``sys.version_info``

### ISSUE THREAD

**Issue by theirix** [pylint-dev/pylint#9153]
### Bug description

```python
Running pylint with Python 3.11 provides the ``no-member`` error even if it is shielded with `sys.version_info` check. The `token.FSTRING_START` is introduced in Python 3.12.


#pylint: disable=missing-module-docstring
import token
import sys

if sys.version_info >= (3, 12):
    print(token.FSTRING_START)


Running pylint with Python 3.12 works fine as expected.
```

### Command used

```shell
pylint x.py
```

### Pylint output

```python
************* Module x
x.py:6:10: E1101: Module 'token' has no 'FSTRING_START' member (no-member)

------------------------------------------------------------------
Your code has been rated at 0.00/10 (previous run: 7.50/10, -7.50)
```

### Expected behavior

There should not be any errors with Python 3.12 or below.

### Pylint version

```shell
pylint 3.0.1
astroid 3.0.1
Python 3.11.4 (main, Jun 20 2023, 16:51:49) [Clang 14.0.0 (clang-1400.0.29.202)]
```

### Additional dependencies

```python
astroid==3.0.1
dill==0.3.7
isort==5.12.0
mccabe==0.7.0
platformdirs==3.11.0
pylint==3.0.1
tomlkit==0.12.1
```
